src/kernel_abi.h

```diff
@@ -497,9 +497,18 @@ struct BaseArch : public wordsize, public FcntlConstants {
   struct shmid64_ds {
     ipc64_perm shm_perm;
     size_t shm_segsz;
-    uint64_t shm_atime_only_little_endian;
-    uint64_t shm_dtime_only_little_endian;
-    uint64_t shm_ctime_only_little_endian;
+    __kernel_time_t shm_atime;
+#if __BITS_PER_LONG != 64
+    unsigned long unused1;
+#endif
```

Comment:
Right, I understand that. But if we add this patch, then when we build rr 64-bit, `X86Arch::shmid64_ds` will contain
````
    __kernel_time_t shm_atime;
    __kernel_time_t shm_dtime;
    __kernel_time_t shm_ctime;
````
which is incorrect since `X86Arch::__kernel_time_t` is only 32 bits.